src/cluster.c

```diff
@@ -3581,20 +3627,49 @@ int clusterNodeCronHandleReconnect(clusterNode *node, mstime_t handshake_timeout
             freeClusterLink(link);
             return 0;
         }
-        node->link = link;
     }
     return 0;
 }
 
-/* Resize the send buffer of a node if it is wasting
- * enough space. */
-int clusterNodeCronResizeBuffers(clusterNode *node) {
+static void resizeClusterLinkBuffer(clusterLink *link) {
      /* If unused space is a lot bigger than the used portion of the buffer then free up unused space.
       * We use a factor of 4 because of the greediness of sdsMakeRoomFor (used by sdscatlen). */
-    if (node->link != NULL && sdsavail(node->link->sndbuf) / 4 > sdslen(node->link->sndbuf)) {
-        node->link->sndbuf = sdsRemoveFreeSpace(node->link->sndbuf);
+    if (link != NULL && sdsavail(link->sndbuf) / 4 > sdslen(link->sndbuf)) {
+        link->sndbuf = sdsRemoveFreeSpace(link->sndbuf);
     }
-    return 0;
+}
+
+/* Resize the send buffer of a node if it is wasting
+ * enough space. */
+static void clusterNodeCronResizeBuffers(clusterNode *node) {
+    resizeClusterLinkBuffer(node->link);
+    resizeClusterLinkBuffer(node->inbound_link);
+}
+
+static void freeClusterLinkOnBufferLimitReached(clusterLink *link) {
+    if (link != NULL && sdsalloc(link->sndbuf) > server.cluster_link_sendbuf_limit_bytes) {
+        freeClusterLink(link);
```

Comment:
I suppose we should add a log message here to explain what happened?

Reply:
Added a WARNING log message, considering this is not expected in normal operation.